presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -384,7 +378,6 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
             // Write to: an existing partition in an existing partitioned table
             if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.APPEND) {
                 // Append to an existing partition
-                checkState(!immutablePartitions);
```

Comment:
undo, this is just defensive programming.

Reply:
I believe this is not good idea to pass immutablePartitions variable through 2 classes just to check it here. 
Will not undo.

Reply:
Please then extract it to separate commit.